### PR TITLE
Improvement/csi 3237 exception catching coverage

### PR DESCRIPTION
**Pull request by ArbelNathan**

### CODE REVIEW

File: controller/array_action/array_mediator_ds8k.py [hunk 1]
```diff
@@ -167,20 +167,12 @@ def _connect(self):
...
-            raise ConnectionError()
-        except exceptions.ClientException as e:
             logger.error(
-                'Failed to connect to DS8K array {}, reason is {}'.format(
-                    self.service_address,
-                    e.details
-                )
-            )
+                'Failed to connect to DS8K array {}, reason is {}'.format(self.service_address, ex.details))
             raise ConnectionError()
```
Comment (oriyarde) [inline controller/array_action/array_mediator_ds8k.py:176]: while you're at it, could you change this to `raise ex`/`raise`, so we won't lose the exception info?

File: controller/array_action/array_mediator_svc.py [hunk 2]
```diff
@@ -502,15 +491,20 @@ def _delete_fcmap(self, fcmap_id, force):
             self.client.svctask.rmfcmap(object_id=fcmap_id, force=force)
         except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
             if not is_warning_message(ex.my_message):
-                logger.warning("Failed to delete fcmap '{0}': {1}".format(fcmap_id, ex))
+                logger.exception("Failed to delete fcmap '{0}': {1}".format(fcmap_id, ex))
```
Comment (oriyarde) [inline controller/array_action/array_mediator_svc.py:494]: `logger.error` is sufficient, no need to log the traceback (same for the other logger calls)

File: controller/tests/array_action/ds8k/test_array_mediator_ds8k.py [hunk 3]
```diff
@@ -706,3 +711,8 @@ def test_expand_volume_extend_not_enough_space_error(self):
         self.client_mock.extend_volume.side_effect = [ClientException("500", message="BE531465")]
         with self.assertRaises(array_errors.NotEnoughSpaceInPool):
             self.array.expand_volume(volume_id="test_id", required_bytes=10)
+
+    def test_expand_volume_extend_raise_other_error(self):
```
Comment (oriyarde) [inline controller/tests/array_action/ds8k/test_array_mediator_ds8k.py:715]: no need for `other` IMO

File: controller/array_action/array_mediator_svc.py (hunk 2 above)
Comment (zingero) [reply controller/array_action/array_mediator_svc.py:494]: +1

File: controller/array_action/array_mediator_ds8k.py (hunk 1 above)
Comment (zingero) [reply controller/array_action/array_mediator_ds8k.py:176]: We catch the `ClientException` exception and raise a different exception, `ConnectionError`.
If we decide to link them together (I'm totally into it), I suggest using `from`:
 ```suggestion
            raise ConnectionError() from ex
```

File: controller/array_action/array_mediator_ds8k.py [hunk 4]
```diff
@@ -274,15 +266,6 @@ def _create_api_volume(self, name, size_in_bytes, space_efficiency, pool_id):
                 )
             )
             raise array_errors.VolumeCreationError(name)
-        except (exceptions.ClientError, exceptions.ClientException) as ex:
-            logger.error(
```
Comment (zingero) [inline controller/array_action/array_mediator_ds8k.py:278]: Why don't we want to print that?

Comment (zingero) [reply controller/array_action/array_mediator_ds8k.py:278]: nvm, got it

File: controller/array_action/array_mediator_ds8k.py [hunk 5]
```diff
@@ -292,11 +275,12 @@ def _extend_volume(self, api_volume, new_size_in_bytes):
         try:
             self.client.extend_volume(volume_id=api_volume.id,
                                       new_size_in_bytes=new_size_in_bytes)
-        except exceptions.NotFound:
-            raise array_errors.ObjectNotFoundError(api_volume.id)
-        except (exceptions.ClientError, exceptions.ClientException) as ex:
+        except exceptions.ClientException as ex:
```
Comment (zingero) [inline controller/array_action/array_mediator_ds8k.py:278]: I'm not 100% like the new implementation. I think it's because the language supports having a couple of `except` statements for one `try` statement and we choose to do it in a different way. What's the reason for this change?

File: controller/array_action/array_mediator_ds8k.py (hunk 1 above)
Comment (oriyarde) [reply controller/array_action/array_mediator_ds8k.py:176]: not sure if `ConnectionInfo` adds any useful info

File: controller/array_action/array_mediator_ds8k.py (hunk 5 above)
Comment (ArbelNathan) [reply controller/array_action/array_mediator_ds8k.py:278]: that way we can catch everything we are looking for and reducing the chance of missing anything, plus, it's more "organized" this way.

File: controller/tests/array_action/ds8k/test_array_mediator_ds8k.py (hunk 3 above)
Comment (ArbelNathan) [reply controller/tests/array_action/ds8k/test_array_mediator_ds8k.py]: done

File: controller/array_action/array_mediator_svc.py (hunk 2 above)
Comment (ArbelNathan) [reply controller/array_action/array_mediator_svc.py]: done

**Review by oriyarde — DISMISSED** [review]
LGTM

File: controller/array_action/array_mediator_ds8k.py (hunk 1 above)
Comment (ArbelNathan) [reply controller/array_action/array_mediator_ds8k.py]: due to the fact we have no handler for `ConnectionInfo` I changed it to `raise ex`

File: controller/array_action/array_mediator_ds8k.py (hunk 5 above)
Comment (zingero) [reply controller/array_action/array_mediator_ds8k.py:278]: We catch "everything" (`ClientException` is everything in this case) anyway, but if @oriyarde and you like this way more than my suggestion, it's fine by me.

Comment (oriyarde) [reply controller/array_action/array_mediator_ds8k.py:278]: https://youtu.be/ussCHoQttyQ

Comment (ArbelNathan) [reply controller/array_action/array_mediator_ds8k.py:280]: done

**Review by oriyarde — APPROVED** [review]
LGTM

**Review by zingero — APPROVED** [review]
